enterprise/templates/enterprise/grant_data_sharing_permissions.html

```diff
@@ -43,10 +47,20 @@
                             data-warning-message="${ messages['warning'] }"
                         />
                         <label for="register-data_sharing_consent">
-                            ${_("I agree to allow {platform_name} to share data about my enrollment, completion and performance in all {platform_name} courses and programs where my enrollment is sponsored by {provider}.*").format(provider=sso_provider, platform_name=platform_name)}
+                            % if not course_id:
+                                ${_("I agree to allow {platform_name} to share data about my enrollment, completion and performance in all {platform_name} courses and programs where my enrollment is sponsored by {provider}.*").format(provider=ec_name, platform_name=platform_name)}
+                            % else:
+                                ${_("I agree to allow {platform_name} to share data about my enrollment, completion and performance in {course_name} with {provider}.*").format(provider=ec_name, course_name=course_name, platform_name=platform_name)}
+                            % endif
                         </label>
                     </div>
                     <input type="hidden" name="csrfmiddlewaretoken" value="${ csrf_token }" />
+                    % if course_id:
+                        <input type="hidden" name="course_id" value="${ course_id }" />
+                    % endif
+                    % if redirect_url:
```

Comment:
Yes. We don't use it in the at-login mode right now, but I suspect we'll need to once enterprise spins out into being a full IDA.